backend/server.js

```diff
@@ -67,28 +71,37 @@ function buildApp (config, store, callback) {
   }
 
   app.use(bodyParser.json());
+  app.use(bodyParser.urlencoded({ extended: true }));
 
   /* Enable OAuth2 authentification only if the database is configured. */
   if (config.database) {
-    return oauth(app, config, function (err) {
-      if (err) return callback('oauth initialization failed');
-      finalizeApp();
-    });
+    if (config.enableOauth) {
+      return oauth(app, config, function (err) {
+        if (err) return callback('oauth initialization failed');
+        finalizeApp();
+      });
+    } else {
+      return userManager(app, config, function (err) {
+        if (err) return callback('user manager initialization failed');
+        finalizeApp();
+      });
+    }
   }
 
-  function finalizeApp () {
+
+  function finalizeApp() {
     addBackendRoutes(app, config, store);
     addOfflineRoutes(app, config, store);
     callback(null, app);
   }
   finalizeApp();
```

Comment:
Here we are calling `finalizeApp()` function three times. Can it be one time only?

Reply:
I haven't looked real deep till now, since its the default flow I don't think we should do any change in it for now.